### PR TITLE
Restored a few error codes and redefined FRAME_ERROR.

### PR DESCRIPTION
**Pull request by janaiyengar**
I'd like to go over this again to ensure we don't leave anything out. In the meanwhile, take a look at the direction I'm pushing in.

### CODE REVIEW

**Comment by martinthomson** [comment]
It sounds like we're going to be discussing this at the interim.  I think that this misses the point of doing a consolidation.

For instance, why would anyone actually *send* HANDSHAKE_TIMEOUT?  For that matter, how would they send it?

**Comment by janaiyengar** [comment]
Good point -- we can remove HANDSHAKE_TIMEOUT. Unless there's a good reason, I'd like to keep the others. I think the errors are useful, and I don't see an advantage to removing the ones that I've listed.

**Comment by martinthomson** [comment]
That was an example, there are quite a number more that fit the same criterion.

**Comment by martinthomson** [comment]
This might help understand where I'm coming from here.  In going through the existing codes, I asked myself the question "what code would I write to handle a CONNECTION_CLOSE/RST_STREAM that contained this error code?"  If the code is the same for two error conditions, the separate codes were redundant.  The other question to ask might be "why would I send a CONNECTION_CLOSE/RST_STREAM containing this error code?"

**Comment by janaiyengar** [comment]
I think I understand where you're coming from. That doesn't cover monitoring though; Google monitors these error codes, and we find it useful operationally. I expect that other server will find it useful too. Happy to air this out at the interim for others to weigh in on.

**Comment by martinthomson** [comment]
@janaiyengar, I had to rebase the `transport_errors` branch, which means that you probably want to rebase this branch too.  `git fetch origin;git checkout revised_errors;git rebase origin/transport_errors` should do the trick.

**Comment by janaiyengar** [comment]
Thanks for the heads-up; rebased.

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -2612,71 +2612,104 @@ CONNECTION_CLOSE or RST_STREAM frame. Error codes share a common code space.
 Some error codes apply only to either streams or the entire connection and have
 no defined semantics in the other context.
 
-INTERNAL_ERROR (0x80000001):
+FRAME_ERROR (0x800000TT):
+
+: An endpoint received a frame that was badly formatted.  For instance, an empty
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: nit: badly -> incorrectly

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -2612,71 +2612,104 @@ CONNECTION_CLOSE or RST_STREAM frame. Error codes share a common code space.
...
+  STREAM frame that omitted the FIN flag, or an ACK frame that has more
+  acknowledgment ranges than the remainder of the packet could carry. The final
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: A better ack example would be an ack packet that indicates previously acked packets are now missing.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -2612,71 +2612,104 @@ CONNECTION_CLOSE or RST_STREAM frame. Error codes share a common code space.
...
-NO_ERROR (0x80000003):
+INVALID_VERSION_NEGOTIATION_PACKET (0x80000106):
+
+: Version negotiation packet is malformed.
+
+PUBLIC_RESET_RECEIVED (0x80000107):
+
+: A public reset packet was received for this connection post-handshake.
+
+TOO_MANY_RTOS (0x80000108):
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: This may be useful, but it does assume one implements a too many RTOs feature.

**Comment by ianswett** [comment]
Is this PR obsolete?

**Pull request closed without merging**